docs/api.md

```diff
@@ -1,12 +1,14 @@
 # API
 
-## Connecting to a network
```

Comment:
@clevinson I saw you started writing how to connect to a network in the `getting-started.md` guide already. I'm not sure it's worth it to put it another time in the API doc.

In this PR, i basically move the same structure into the `getting-started.md` file. This file will only talk about the "interacting with your node" part.